common/lib/util/defaults.js

```diff
@@ -39,6 +39,13 @@ Defaults.version          = '1.2.8';
 Defaults.libstring        = Platform.libver + '-' + Defaults.version;
 Defaults.apiVersion       = '1.2';
 
+var agent = 'ably-js/' + Defaults.version;
```

Comment:
Given that this is supposed to replace lib, seems very odd we'd include less information than that. From the libstring I can tell whether the library is browser, react-native, nativescript etc., this drops that information. If this is to help us decide whether to drop support for things, that's moving in the wrong direction, surely.

Is this why you left the lib param in? If so I'm unconvinced - having two platfrom version strings in different formats where the information you need is spread across both of them is not great, if we're going to switch formats we should actually make the new one a replacement for the old, if we're not going to do that we shouldn't switch formats

Reply:
I've now updated the agent param to make use of the existing libstring (2b44cdf), and removed the lib param completely (e5c8527).

Reply:
I would like to explore this further as I think the updates that @owenpearson has made in response to @SimonWoolf's comments may be too simplistic. We only get one shot at this and I want to aim for maximum flexibility - so, as such, it would appear to me that a discrete agent product component purely to indicate the `ably-js` source codebase version may be more helpful, where it's then augmented by an additional agent product component(s) telling us about the runtime environment and/or wrapper.

I'm going to accumulate thoughts on this in [this internal document](https://docs.google.com/document/d/1duUdRknI1QNkanASgkTYo19LGXQIeT8oFlXS7TCU-Ek/edit?usp=sharing) for now.

Reply:
Looking at Node's [os module](https://nodejs.org/api/os.html#os_os), we could consider adding [platform()](https://nodejs.org/api/os.html#os_os_platform)`/`[version()](https://nodejs.org/api/os.html#os_os_version), for example.